### DISCUSSION TITLE
Extracting Water Surface Elevation Data?

### DISCUSSION THREAD

**Discussion by aepoc** [DualSPHysics/DualSPHysics#246] [DUALSPHYSICS_V5.4_BETA]
Hey there,

I am investigating oscillating water columns and would like to estimate the natural frequency of various caisson geometries.  I believe I can do this using Fourier Transforms in NumPy or similar.  

My question:  Is there a way to export particle elevation data so that it can be processed using FFTs in Python?  My simulations are quite large (70 to 80 million particles) so I will need to somehow select only the particles I am interested in...  FFTs on 80 million particles might take some time.

Is there any other way to use Dualsphysics to determine a natural frequency of any given geometry?

Thank you!

**Comment by alexbexe** [comment]
Determining natural frequency is just a postprocessing step you can do on your own using the time series of the motion of the device.

FloatingInfo already gives you the 6DoF for each object

Regards

**Reply by aepoc** [reply]
Thank you Alex

Can you explain further?  Is this a postprocessing step that I can do with a batch file or from the XML?  If not, where can I find the floatingInfo that I need for the calculations?

**Comment by alexbexe** [comment]
FloatingInfo is a post-processing tool that you can execute adding the command lines in the BATCH file.
You can find many examples in the release

Regards

**Comment by hojjatanvari** [comment]
Hello aepoc,
I am working on a similar project, but my problem is not this. 
You can get modal displacements, velocities, and forces by defining the boundary as a floating body in `<floatings>`:
```
<floating mkbound="10" rhopbody="2700">
                <translationDOF x="0" y="0" z="0" />
                <rotationDOF x="0" y="0" z="0" />
</floating>
```

And then inputting mode shapes, and their corresponding modal mass and modal stiffness in a csv file:
```
<hydroelastics>
                <hydroelastic mkbound="20">
                    <modefile value="flexible_wall_data.csv" comment="File with Modal Inertia, Modal Stiffness, mode position data and mode shapes" />
                </hydroelastic>
</hydroelastics>
```

Finally, you will get the wet natural frequencies from the FFT of modal displacements.

My problem: There is a significant error between this method of calculating wet natural frequencies and experimental or ABAQUS results. I think something is wrong with the input modal data, but I don't know what. Please let me know if you get a valid result.
